crates/oxc_parser/src/js/statement.rs

```diff
@@ -37,9 +37,10 @@ impl<'a> Parser<'a> {
                     let stmt = self.parse_export_declaration()?;
                     statements.push(stmt);
                 }
-                Kind::At => {
+                Kind::At if self.at_decorator_before_export() => {
```

Comment:
I think we need to fix / workaround with this, it's causing perf hit on all top level statements. 

We need to consume the decorator and then check what token is next.

Is it a good idea to save the start span as a state inside eat_decorators?

Reply:
Oh sure. We don't need this if doing the babel way.